repo/config/init.go

```diff
@@ -93,15 +93,22 @@ func datastoreConfig() (Datastore, error) {
 }
 
 // identityConfig initializes a new identity.
-func identityConfig(out io.Writer, nbits int) (Identity, error) {
+func identityConfig(out io.Writer, nbits, keyType int) (Identity, error) {
 	// TODO guard higher up
 	ident := Identity{}
-	if nbits < 1024 {
-		return ident, errors.New("Bitsize less than 1024 is considered unsafe.")
+
+	switch keyType {
```

Comment:
Basically you need a default handler that errors out with something helpful for the user.